zhmc_prometheus_exporter/zhmc_prometheus_exporter.py

```diff
@@ -2407,7 +2421,26 @@ def main():
                                    hmc_info['api-minor-version'])
                 client = zhmcclient.Client(session)
                 hmc_features = client.consoles.console.list_api_features()
+
+                # Determine list of CPCs to export, as specified in 'cpcs'.
+                # 'cpcs' is optional and defaults to all managed CPCs.
                 cpc_list = client.cpcs.list()
+                cpc_names = [cpc.name for cpc in cpc_list]
+                if not cpc_list:
+                    raise ImproperExit(
+                        "This HMC does not manage any CPCs.")
+                if yaml_cpcs is not None:
+                    ne_cpc_names = []
+                    for cpc_name in yaml_cpcs:
+                        if cpc_name not in cpc_names:
+                            ne_cpc_names.append(cpc_name)
```

Comment:
Minor: The loop could be replaced with a list comprehension.

Reply:
Done